github_jira_sync_app/main.py

```diff
@@ -264,14 +269,31 @@ async def bot(request: Request, payload: dict = Body(...)):
         new_issue = jira.create_issue(fields=issue_dict)
         existing_issues.append(new_issue)
 
-        if settings["add_gh_comment"]:
-            issue.create_comment(
-                gh_comment_body_template.format(jira_issue_link=new_issue.permalink())
-            )
+        synced_label_absent_in_repo = False
+
+        if settings["add_gh_synced_label"]:
+            try:
+                repo.get_label(gh_synced_label_name)
```

Comment:
what happens if we do not try to get a label?

can we just rely on exception ?

Reply:
If the label doesn't exist the method throws `github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "documentation_url": "https://docs.github.com/rest/issues/labels#get-a-label", "status": "404"}`.

The alternative is something like `if "synced-to-jira" in [l.name for l in repo.get_labels()]:` which looks a bit overcomplicated and unnecessary to me.

Reply:
GitHub has rate limits on number of requests sent from an app. I try to limit the number of API calls as much as possible.

so, in both cases we get an exception (if I read your message right), but if we fail we sent 1 call, if succeed we send 2 calls. 

If we remove `repo.get_label(gh_synced_label_name)`, then it is a single call in both cases.
Is that correct?

Reply:
Sorry, I misunderstood what you meant.
Yes, it is possible to have only one API call and avoid the verification.
When we add a nonexistent label what we get is that an exception will be thrown from the library.
Looking at [the reference](https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#add-labels-to-an-issue) it will either return a 404 resulting in a `GitHubException.UnknownObjectException` or a 422 resulting in `GithubException.GithubException`. From the description it should be the latter, but i don't think I have an easy setup to test that.

Reply:
I have an infra to test on staging. Please apply the change and I can test

Reply:
Pushed the temporary solution with the 422 scenario.
The resulting body and error code should then be put in the test template `tests/unit/url_responses/github_responses_synced_label_notfound.yaml`